cfme/tests/intelligence/reports/test_menus.py

```diff
@@ -171,3 +169,58 @@ def test_rbac_move_custom_report(
             type=folder, subtype=subfolder, menu_name=get_custom_report.menu_name
         )
         assert rbac_report.exists
+
+
+@pytest.mark.tier(1)
+@pytest.mark.ignore_stream("5.10")
+@pytest.mark.meta(automates=[1676638, 1731017])
+@pytest.mark.parametrize("group", GROUPS)
+def test_reports_menu_with_duplicate_reports(appliance, request, group, report_menus):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        initialEstimate: 1/10h
+        startsin: 5.9
+        setup:
+            1.Create a custom report or copy an existing report.
+            2. Move the custom report to 'Tenants > Tenant Quotas' menu in user's current group
+        testSteps:
+            1. See if the report is available under 'Tenants > Tenant Quotas'.
+            2. Delete the custom report.
+            3. See if the report is available under 'Tenants > Tenant Quotas'.
+            4. Add a new report with the same menu_name.
+            5. See if the report is available under 'Tenants > Tenant Quotas'.
+        expectedResults:
+            1. Report must be visible.
+            2.
+            3. Report must not be visible.
+            4.
+            5. Report must not be visible.
+    Bugzilla:
+        1731017
+        1676638
+    """
+    # Copy a report the first time and move it
+    custom_report_1 = appliance.collections.reports.instantiate(
+        type="Tenants", subtype="Tenant Quotas", menu_name="Tenant Quotas"
+    ).copy()
+    folder, subfolder = "Tenants", "Tenant Quotas"
+    report_menus.move_reports(group, folder, subfolder, custom_report_1.menu_name)
+
+    # instantiate the expected report and check for its existence
+    expected_report = appliance.collections.reports.instantiate(
+        type=folder, subtype=subfolder, menu_name=custom_report_1.menu_name
+    )
+    assert expected_report.exists
+
+    # delete the first report and check if the expected report exists
+    custom_report_1.delete()
```

Comment:
Suggestion:
You can use `delete()` operation inside (`request.addfinalizer(custom_report_2.delete)`)

Reply:
- I think this is the part of test case. If test fails to delete the ```custom_report_1``` then this test should fail.
- So according to me, we don't need ```request.addfinalizer(custom_report_1.delete)```

What do you think @valaparthvi ?

Reply:
Hmm, that does seem like a valid point :sweat_smile:

Reply:
I think then finalizer not is require for copied(`custom_report_2`) report, simply `delete()` will do the needful.?
@valaparthvi can you please confirm this .? Just Curious :smile:

Reply:
Deleting `custom_report_1` was a part of the test, but deleting `custom_report_2` is not, so it's better to delete `custom_report_2` as a part of the cleanup.